rfcs/text/0000-matrix-service.md

```diff
@@ -0,0 +1,406 @@
+| Comparison Filter | {<br />   `op`: one of `[ =, !=, >, <, >=, <=, in ]`,<br />   `field`: a metadata field,<br />   `value`: string \| int \| list <br />} |
+| Logical Filter    | {<br />   `op`: one of `[ and, or, not ]`,<br />   `value`: array of >=2 filter objects `if op==and|or`, filter object `if op==not` <br />} |
+
+Comparison Filters specify a condition on a metadata field and Logical Filters implement `and`, `or` and `not` logic. Note that the `value` field of a Logical Filter accepts a base filter object and is the mechanism for nesting chains of AND/OR statements. For a Comparison Filter's `field` parameter, the list of accepted metadata filters is available at the `/filters` endpoint. More information about a filter such as a description and a set of valid values can be found at the `/filters/<filter>` endpoint. See [Auxiliary Endpoints](#auxiliary-endpoints) for more details on these.
+
+Ex. Select all full length cells:
+
+```
+"filter": {
+  "op": ">=",
+  "value": "full length",
+  "field": "library_preparation_protocol.end_bias"
+}
+```
+
+Ex. Select all cells from the "Single cell transcriptome analysis of human pancreas" project with at least 3000 genes detected:
+
+```
+"filter": {
+  "op": "and",
+  "value": [
+    {
+      "op": "=",
+      "value": "Single cell transcriptome analysis of human pancreas",
+      "field": "project.project_core.project_short_name"
+    },
+    {
+      "op": ">=",
+      "value": 3000,
+      "field": "genes_detected"
+    }
+  ]
+}
+```
+
+#### Fields
+
+Users may supply a list of metadata `field`s to be exported with their generated matrix. For `loom` formats, metadata is made available within the `loom` file itself via an adjacent attribute matrix; `mtx` and `csv` formats provide a separate CSV file to serve user requested metadata. The full list of available metadata fields is available at the [/fields endpoint](https://matrix.data.humancellatlas.org/v1/fields). More information about a specific metadata field such as a description and a set of valid values is available at the `/fields/<field>` endpoint. See [Auxiliary Endpoints](#auxiliary-endpoints) for more details on these.
+
+#### Format
+
+The API generates expression matrices in three formats: `loom`, `mtx` and `csv`. From the [loom](http://loompy.org/) documentation page, the format is "an efficient file format for large omics datasets", enabling the storage of large expression matrices with metadata and also supports many popular data processing programming languages. For these reasons, `loom` is the service's default and most performant format. The `mtx` format adheres to the [10x feature-barcode matrix specification](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices) to enable compatibility with `scanpy` and `seurat` 10X methods. `csv` is supported as a raw text format for which performance may not be optimized for.
```

Comment:
- The loom description is already written above so could delete here? (line 43)
- scanpy and seurat aren't 10x methods? 
- can delete "for" in the last sentence.

Reply:
- Would prefer to keep the section self-sufficient
- Refers to 10x methods from `scanpy` and `seurat`. Will reword
- :+1: